ARROW-16626: [C++] Name the C++ streaming execution engine (#13207)
apache/arrow: docs/source/cpp/overview.rst

**Kernels** are specialized computation functions running in a loop over a
given set of datums representing input and output parameters to the functions.

> **Acero** is a streaming execution engine that allows computation to be expressed

Reviewer: ```suggestion
**Acero** (pronounced AH-serr-oh) is a streaming execution engine that allows computation to be expressed
```

> **Acero** (pronounced [aˈsɜɾo] / ah-SERR-oh) is a streaming execution engine that allows

Reviewer 2: ```suggestion
**Acero** (pronounced [aˈsɜɹo] / ah-SERR-oh) is a streaming execution engine that allows
```

This is fantastic, thanks! 

One very very minor update to the r sound in IPA. /ɾ/ is a flap (what ts and ds are frequently reduced to), and the original /r/ is common in quick transcriptions of English, but not technically accurate, since it's a trilled r like in (many dialects of) Spanish.

Everything else looks good to me!

> **Acero** (pronounced AH-serr-oh) is a streaming execution engine that allows computation to be expressed

Reviewer 2: ```suggestion
**Acero** (pronounced ʌˈsɜroʊ) is a streaming execution engine that allows computation to be expressed
```

Suggestion to use IPA if we are providing a pronunciation guide (also the stress is the second syllable, yeah?)

Reviewer: I think you are right about the second syllable being stressed. Never got past linguistics 101.

IPA is more precise, but is it recognizable to most people? I'd personally have to copy and paste the IPA into a reader to get it.

Author: If we're modeling after the [Spanish word](https://en.wiktionary.org/wiki/%C3%A1cero) then it seems we would want `ˈa.se.ɾo` but I agree I've been pronouncing it internally as `ʌˈsɜroʊ` (`Uh-serr-oh`).

I also agree with Will that IPA is somewhat intimidating to the layman.  Perhaps we can do both versions?

Reviewer: Pronunciation is hard; the word exists across many languages and dialects. This is what it sounds like to me: `[ˈa.sɜ.ɾo]` 🤷 
```suggestion
**Acero** (typically pronounced [ˈa.sɜ.ɾo] / AH-serr-oh) is a streaming execution engine that allows computation to be expressed
```

Or maybe just don't provide a guide 🤦

Reviewer 2: This is the (former) linguist in me showing, but TBF, the other systems of pronunciation all rely on some pretty heavy base dialect being known (e.g. some are for North American English, some are for UK English and they don't necessarily overlap!) to the point that they are almost meaningless at conveying the pronunciation outside of a particular group of English speakers. That being said, pragmatically, if we have both that's fine, so long as the IPA is one of them 😝 

You'ld still want to move the stress to the second syllable in both I think  `[a.ˈsɜ.ɾo] / ah-SERR-oh` right? Also, to 🚲 🏠 this even more, I vote we don't put the syllable breaks in the IPA (it's toootally kosher to leave them out) so we can _gloss over_ the complexities in syllabification rules for English rhotics (and then the fact that the two guides differ in which syllable gets the `r` isn't noticeable 🙈 )
